content/shmem_team_alltoall.tex

```diff
@@ -0,0 +1,83 @@
+\apisummary{
+    Transfers fixed-size blocks of data from each \ac{PE} in a team
+    to all \acp{PE} in the team.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void @\FuncDecl{shmem\_team\_alltoall8}@(shmem_team_t team, void *dest, const void *source, size_t nelems);
+void @\FuncDecl{shmem\_team\_alltoall16}@(shmem_team_t team, void *dest, const void *source, size_t nelems);
```

Comment:
There was some question during the 10-4 WG call about why to add 8/16 versions of this function for teams when the non-teams alltoall only provides 32/64. Suggestion is to have a separate issue and PR for changing types for collectives and make changes across the board to provide more consistent, portable interface.

Reply:
Roger. I can split this off as a separate PR for expanding all the all-to-all routines with the 8/16B sizes. As to why, I think it's mostly an issue of completeness. Admittedly, the primary case for all-to-all will use the 32B and 64B routines, so maybe it's just not necessary. IIRC, the original Cray SHMEM extension for all-to-all was just a byte mover, `shmemx_alltoall`, and we did make some assumptions about this byte-level generality.